cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -518,3 +525,13 @@ type Arg struct {
 type EnvVar struct {
 	v1.EnvVar
 }
+
+// EncryptionAlgorithmType can define an asymmetric encryption algorithm type.
+type EncryptionAlgorithmType string
+
+const (
+	// EncryptionAlgorithmECDSA defines the ECDSA encryption algorithm type.
+	EncryptionAlgorithmECDSA EncryptionAlgorithmType = "ecdsa"
+	// EncryptionAlgorithmRSA defines the RSA encryption algorithm type.
+	EncryptionAlgorithmRSA EncryptionAlgorithmType = "rsa"
```

Comment:
https://github.com/golang/go/blob/ace1494d9235be94f1325ab6e45105a446b3224c/src/crypto/x509/x509.go#L263-L268 

Is it better to use `RSA` and `ECDSA` uppercase to the same as above/the original? Not sure if this must be or not.

Reply:
the k8s API docs recommend uppercase for abbreviations in field names, but i don't remember if there is rule for field values.

uppercase seems better to me actually.

Reply:
i had a look at what we have at https://git.k8s.io/api and it feels like there isn't a strict rule for constant abbreviation values, so there are two types:
- `cpu` all lowercase abbreviation
- `VIP` all uppercase  abbreviation
the only rule about abbreviations is to have them in consistent case e.g. don't do `Vip`, must be `vip` or `VIP`.

https://github.com/kubernetes/community/blob/ce04e02575b252d6d7baa749b74c287fd7d4a5a2/contributors/devel/sig-architecture/api-conventions.md

Reply:
updated!

Reply:
/lgtm